roscpp_azure_iothub/src/ros_azure_iothub_cpp_node.cpp

```diff
@@ -244,25 +292,56 @@ void topicCallback(const topic_tools::ShapeShifter::ConstPtr& msg,
     parser.deserializeIntoFlatContainer( topic_name, Span<uint8_t>(buffer), &flat_container, 100);
     parser.applyNameTransform( topic_name, flat_container, &renamed_values );
 
-    ROS_INFO("--------- %s ----------", topic_name.c_str() );
-    sendMsgToAzureIoTHub(topic_name.c_str(), deviceHandle);
-    for (auto it: renamed_values)
+    // Send info to IoTHub via reported properties 
+    if (!(std::find(topicsToReport.begin(), topicsToReport.end(), topic_name.c_str()) == topicsToReport.end()))
     {
-        const std::string& key = it.first;
-        const Variant& value   = it.second;
-        char _buffer [256] = {0};
-        snprintf(_buffer, sizeof(_buffer), " %s = %f", key.c_str(), value.convert<double>());
-        ROS_INFO("%s",_buffer);
-        sendMsgToAzureIoTHub(_buffer, deviceHandle);
+        ROS_INFO("Sending message from %s to IoTHub via reported properties ", topic_name.c_str()); 
+
+        std::string topic_msg = "";
+        bool first = true;
+
+        for (auto it: renamed_values)
+        {
+            const std::string& key = it.first;
+            const Variant& value   = it.second;
+            char _buffer [256] = {0};
+            snprintf(_buffer, sizeof(_buffer), "%s = %f", key.c_str(), value.convert<double>());
```

Comment:
It looks like the original code always assumed the variable was double instead of introspecting the variant - is that true?

Reply:
It did for the flat_container but not the renamed_values .

Reply:
snprintf will return the count of characters required; we should handle the case when there is an overflow so we don't drop values. Let's create a bug for this and fix it in another checkin.

Reply:
It looks like a bug - let's create a bug for this and do it in another checkin.